server/protocols/jmap/src/main/java/org/apache/james/jmap/model/mailbox/Rights.java

```diff
@@ -213,6 +217,36 @@ public MailboxACL toMailboxAcl() {
             .reduce(MailboxACL.EMPTY, union);
     }
 
+    public Optional<Boolean> mayReadItems(Username username) {
+        return containsRight(username, Right.Read);
+    }
+
+    public Optional<Boolean> mayAddItems(Username username) {
+        return containsRight(username, Right.Insert);
+    }
+
+    public Optional<Boolean> mayCreateChild(Username username) {
+        return UNSUPPORTED;
```

Comment:
why?

Reply:
Because the semantic of JMAP Right as we defined it do not allow me to express this may property.

Thus I'm returning empty, and let it being resolved by the caller.